assets/src/domain/eventEditor/ui/tickets/ticketPriceCalculator/fields/TicketPrice.tsx

```diff
@@ -16,12 +17,12 @@ const TicketPriceField: React.FC<TicketPriceFieldProps> = (props) => {
 	const setValue: BFP['setValue'] = useCallback((value) => updateTicketPrice(value), [updateTicketPrice]);
 
 	return (
-		<div className='ee-ticket-price-field'>
-			<div className='ee-ticket-price-field__before'>{beforeAmount}</div>
-			<BaseField {...props} getValue={getValue} setValue={setValue} name={'ticket.price'} />
-			<div className='ee-ticket-price-field__after'>{afterAmount}</div>
-		</div>
+		<TicketPriceField
+			after={afterAmount}
+			before={beforeAmount}
+			field={<BaseField {...props} getValue={getValue} setValue={setValue} name={'ticket.price'} />}
+		/>
```

Comment:
great idea moving this into a separate component, but let's give it a more generic name so that it can be applied to any money input

maybe call it `MoneyField` and have this new component:

- add optional props for currency details, and/or pass a currency object/type (dunno if we have one now or not)

- if the above is not provided then pull in the currency details from the currency config for determining whether the currency sign appears before or after the amount input

- also add prop for setting whether to display a currency sign or percentage sign, maybe named `isPercent` with default value of `false`

- utilize the actual input as a child component (as opposed to using a prop) ie:

```js
<MoneyField currency={currency}>
	<TicketPriceField {...props} getValue={getValue} setValue={setValue} name={'ticket.price'} />
</MoneyField>
```

then we can use this even for wrapping the ticket price inline edit field in the ticket card details

- plz also see my comments here: https://github.com/eventespresso/event-espresso-core/pull/2869/files#r430626075

Reply:
currency object can also be pulled from `useConfig`, which is what I did, since it is a common prop and it won't render so often to cause perf issues

Reply:
I noticed that we also have `PriceTypeSign` which might overlap a bit with what you have described above, the only difference is that it is requiring `priceType` with `isPercent`.